core/src/main/java/com/sequenceiq/cloudbreak/service/publicendpoint/GatewayPublicEndpointManagementService.java

```diff
@@ -276,4 +287,16 @@ private String getEndpointNameForStack(Stack stack) {
         return loadBalancerPersistenceService.findByStackId(stack.getId()).stream()
             .map(LoadBalancer::getEndpoint).collect(Collectors.toSet());
     }
+
+    private Optional<LoadBalancer> selectLoadBalancer(Set<LoadBalancer> loadBalancers) {
+        Optional<LoadBalancer> loadBalancerOptional = loadBalancers.stream()
```

Comment:
dulplicated logic

Reply:
Per Mihaly's comment below, I've removed this and the other `selectLoadBalancer` method and replaced them with a more generalized method in `LoadBalancerConfigService`.